frontend/src/utils/error-handler.ts

```diff
@@ -0,0 +1,46 @@
+import posthog from "posthog-js";
+import toast from "react-hot-toast";
+import { jsx as _jsx } from "react/jsx-runtime";
+import { ErrorToast } from "#/components/shared/error-toast";
+import { handleStatusMessage } from "#/services/actions";
+
+interface ErrorDetails {
+  message: string;
+  source?: string;
+  metadata?: Record<string, unknown>;
+  msgId?: string;
+}
+
+export function logError({ message, source, metadata = {} }: ErrorDetails) {
+  const error = new Error(message);
+  posthog.captureException(error, {
+    error_source: source || "unknown",
+    ...metadata,
+  });
+}
+
+export function showErrorToast({
+  message,
+  source,
+  metadata = {},
+}: ErrorDetails) {
+  logError({ message, source, metadata });
+  toast.custom((t: { id: string }) =>
+    _jsx(ErrorToast, { id: t.id, error: message }),
```

Comment:
When I tried this the styling didn't work, so my PR suggestion reverts this to the current behavior, `toast.error(...)`, and we can change the display later.